omod/src/main/java/org/openmrs/module/bedmanagement/rest/resource/AdmissionLocationResource.java

```diff
@@ -118,16 +115,6 @@ public Model getGETModel(Representation rep) {
 		return modelImpl;
 	}
 	
-	@PropertyGetter("bedLayouts")
-	public Object getBedLayouts(AdmissionLocation admissionLocation) throws Exception {
-		List<BedLayout> bedLayouts = admissionLocation.getBedLayouts();
-		List<SimpleObject> ret = new ArrayList<SimpleObject>();
-		for (BedLayout bedLayout : bedLayouts) {
-			ret.add(getBedLayout(bedLayout));
-		}
-		return ret;
-	}
-	
```

Comment:
#These custom property getters seem to be the fundamental issue, in that they return SimpleObjects, which are just Maps, rather than objects that the converter chains (eg. full rep, custom rep) know how to convert in different ways.  When using the SimpleObjects like this, the representation is effectively hard-coded and not flexible.  I have removed this in favor of just getting the BedLayout objects, and then ensuing BedLayout has a resource that can inform how it should be converted.

Reply:
What happens to the API? I am assuming that the API contract is now changed? If so, this will likely break existing Bahmni feature, unless the frontend is modified appropriately as well.

Reply:
No, this has not changed the API contract.  I ensured the same default and full representations contained the same fields (though there are now more fields than before).